File: src/main/java/com/carenode/entity/Diagnostic.java
package com.carenode.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "diagnostics")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Diagnostic {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "encounter_id")
    private Encounter encounter;

    @Column(name = "image_r2_key")
    private String imageCloudinaryKey;

    private String modelName;

    private String prediction;

    @Column(precision = 5, scale = 4)
    private BigDecimal confidenceScore;

    @Column(columnDefinition = "jsonb")
    private String rawOutput;

    private LocalDateTime createdAt;
}

File: app/model/Diagnostic.js
// app/model/Diagnostic.js
import { Model } from '@nozbe/watermelondb';
import { field, text, date, relation, json } from '@nozbe/watermelondb/decorators';

export default class Diagnostic extends Model {
  static table = 'diagnostics';
  static associations = {
    encounters: { type: 'belongs_to', foreignKey: 'encounter_id' },
  };

  @relation('encounters', 'encounter_id') encounter;
  @field('image_r2_key') imageR2Key;
  @field('model_name') modelName;
  @field('prediction') prediction;
  @field('confidence_score') confidenceScore;
  @json('raw_output', (json) => json) rawOutput;
  @date('created_at') createdAt;
}